diff --git a/utils.h b/utils.h
index 2a84337..074f852 100644
--- a/utils.h
+++ b/utils.h
@@ -7,11 +7,20 @@
 #include <vector>
 #include <random>
 #include <thread>
+#include <cstdlib>
 
 //
 // CLI argument parsing
 //
 
+inline static std::string get_data_home() {
+    char* xdg_data_home = std::getenv("XDG_DATA_HOME");
+    if (xdg_data_home) return xdg_data_home;
+    std::string home = std::getenv("HOME");
+    return home + "/.local/share";
+}
+
+
 // The default parameters
 struct gpt_params {
     int32_t seed      = -1; // RNG seed
@@ -28,7 +37,7 @@ struct gpt_params {
 
     int32_t n_batch = 8; // batch size for prompt processing
 
-    std::string model = "ggml-alpaca-7b-q4.bin"; // model path
+    std::string model = get_data_home() + "/alpaca-cpp/ggml-alpaca-7b-q4.bin"; // model path
     std::string prompt;
 
     bool use_color = true; // use color to distinguish generations and inputs
